Core/src/V2/Iam.php

```diff
@@ -0,0 +1,249 @@
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    private $gapic;
+
+    /**
+     * @var string
+     */
+    private $resource;
+
+    /**
+     * @var array
+     */
+    private $policy;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @param RequestHandler $requestHandler
+     * @param Serializer $serializer The serializer instance to encode/decode messages.
+     * @param string $resource
+     * @access private
+     */
+    public function __construct(
+        RequestHandler $requestHandler,
+        Serializer $serializer,
+        $gapic,
+        $resource
+    ) {
+        $this->requestHandler = $requestHandler;
+        $this->serializer = $serializer;
+        $this->gapic = $gapic;
+        $this->resource = $resource;
+    }
+
+    /**
+     * Get the existing IAM policy for this resource.
+     *
+     * If a policy has already been retrieved from the API, it will be returned.
+     * To fetch a fresh copy of the policy, use
+     * {@see Google\Cloud\Core\Iam\Iam::reload()}.
+     *
+     * Example:
+     * ```
+     * $policy = $iam->policy();
+     * ```
+     *
+     * @param  array $options Configuration Options
+     * @param  int   $options['requestedPolicyVersion'] Specify the policy version to
+     *     request from the server. Please see
+     *     [policy versioning](https://cloud.google.com/iam/docs/policies#versions)
+     *     for more information.
+     * @return array An array of policy data
+     */
+    public function policy(array $options = [])
+    {
+        if (!$this->policy) {
+            $this->reload($options);
+        }
+
+        return $this->policy;
+    }
+
+    /**
+     * Set the IAM policy for this resource.
+     *
+     * Bindings with invalid roles, or non-existent members will raise a server
+     * error.
+     *
+     * Example:
+     * ```
+     * $policy = [
+     *      'bindings' => [[
+     *          'role' => 'roles/editor',
+     *          'members' => ['user:user:test@example.com'],
+     *      ]]
+     * ];
+     * $iam->setPolicy($policy);
+     * ```
+     * ```
+     * $oldPolicy = $iam->policy();
+     * $oldPolicy['bindings'][0]['members'] = 'user:test@example.com';
+     *
+     * $policy = $iam->setPolicy($oldPolicy);
+     * ```
+     *
+     * @param  array|PolicyBuilder $policy The new policy, as an array or an
+     *         instance of {@see Google\Cloud\Core\Iam\PolicyBuilder}.
+     * @param  array $options Configuration Options
+     * @return array An array of policy data
+     * @throws \InvalidArgumentException If the given policy is not an array or PolicyBuilder.
+     */
+    public function setPolicy($policy, array $options = [])
+    {
+        if ($policy instanceof PolicyBuilder) {
+            $policy = $policy->result();
+        }
+
+        if (!is_array($policy)) {
+            throw new \InvalidArgumentException('Given policy data must be an array or an instance of PolicyBuilder.');
```

Comment:
nit: import `InvalidArgumentException` (and update all `@throws` to remove the leading `\`)